src/react-render-perf/lesson-03/solution/swatch.tsx

```diff
@@ -0,0 +1,30 @@
+import {useState} from "react";
+
+import {Color, toCssColor} from "../../shared/color";
+import {colorPickerEventEmitter} from "./color-picker-event-emitter";
+
+type Props = {
+    color: Color;
+    size: number;
+};
+
+export function Swatch({color, size}: Props) {
+    const [isSelected, setIsSelected] = useState(false);
+
+    colorPickerEventEmitter.onColorChange(toCssColor(color), setIsSelected);
```

Comment:
hrm we're just calling this inline in the render function? Should it be inside a `useEffect`?

Reply:
Good catch.